src/utils/system_utils/src/system_utils.c

```diff
@@ -67,6 +68,52 @@ const char* ADUC_SystemUtils_GetTemporaryPathName()
     return env;
 }
 
+/**
+ * @brief mktemp() implementation
+ *
+ * mktemp() is generally considered unsafe as the temporary names can be easy to guess, and there are
+ * a limited number of combinations.
+ *
+ * However, for unit tests, it can be useful, so it's highly recommended that this method is only
+ * used for unit tests.
+ *
+ * @param tmpl The last six characters XXXXXX are replaced with a string that makes the filename unique.
+ * @return Returns @param tmpl.
+ */
+char* ADUC_SystemUtils_MkTemp(char* tmpl)
+{
+    const size_t len = strlen(tmpl);
```

Comment:
it'd be safer if char len of the template string were passed in.

Reply:
that's not how mktemp works, so I didn't want to deviate, but I agree in principle.